roles/ceph-mgr/tasks/mgr_modules.yml

```diff
@@ -41,3 +41,13 @@
   with_items: "{{ ceph_mgr_modules }}"
   delegate_to: "{{ groups[mon_group_name][0] }}"
   when: (item in _disabled_ceph_mgr_modules or _disabled_ceph_mgr_modules == [])
+
+- name: check prometheus plugin port
+  command: "{{ container_exec_cmd_mgr | default('') }} ceph --cluster {{ cluster }} config get mgr. mgr/prometheus/server_port"
+  register: ceph_mgr_prometheus_plugin_actual_port
+
+- name: set prometheus plugin port
+  command: "{{ container_exec_cmd_mgr | default('') }} ceph --cluster {{ cluster }} config set mgr mgr/prometheus/server_port {{ ceph_mgr_prometheus_plugin_port }}"
+  delegate_to: "{{ groups[mon_group_name][0] }}"
+  when: ceph_mgr_prometheus_plugin_actual_port.stdout != ceph_mgr_prometheus_plugin_port
+  notify: restart ceph mgrs
```

Comment:
You don't need to restart the mgrs. You can only restart the prometheus module.

Reply:
@dsavineau You mean disable\enable prometheus plugin? I couldn't find any other ways to restart one module. And, should I make a separate handler to restart the module and add it to ceph-handler/handlers/main.yml or create handlers dir in the ceph-mgr role?